src/ledger/utils.ts

```diff
@@ -101,13 +102,28 @@ function compareTransactions(
   return first.outcome.ledgerVersion < second.outcome.ledgerVersion ? -1 : 1
 }
 
+function hasLedgerVersions(connection: Connection, min: number, max: number | undefined): Promise<boolean> {
```

Comment:
This function should be `async` if it returns a promise. Also, I generally prefer `async`/`await` syntax to `.then()` syntax, but its up to you.